File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key = lambda x : x[1])
        cnt = 0
        first = intervals[0][1]
        for interval in intervals[1:]:
            if first > interval[0]:
                cnt += 1
            else:
                first = interval[1]
        return cnt

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, Comparator.comparingInt(a -> a[1]));
        int first= intervals[0][1];
        int cnt = 0;
        for(int i =1; i < intervals.length; i++){
            if(first > intervals[i][0]) cnt++;
            else first = intervals[i][1];
        }
        return cnt;
    }
}